scheduler/src/cook/mesos/util.clj

```diff
@@ -339,21 +345,73 @@
 (def ^:const job-states #{"running" "waiting" "completed"})
 (def ^:const instance-states #{"success" "failed"})
 
+
+(defn job-submit-time-between
+  "Returns true if the job-ent's :job/submit-time is between start-ms (inclusive) and
+  end-ms (exclusive)"
+  [{:keys [^Date job/submit-time]} ^long start-ms ^long end-ms]
+  (let [submit-ms (.getTime submit-time)]
+                    (and (<= start-ms submit-ms)
+                         (< submit-ms end-ms))))
+
+;; get-jobs-by-user-and-state-and-submit is a bit opaque
+;; because it is reaching into datomic internals. Here is a quick explanation.
+;; seek-datoms provides a pointer into the raw datomic indices
+;; that we can then seek through. We set the pointer to look
+;; through the avet index, with attribute :job/state, seek to
```

Comment:
This says `avet`, but the code below seems to be using `vaet`. Do we mean to be using `vaet`? If so, it would be good to explain why `vaet` is a better choice here.

Reply:
Comment fixed. See description a top of PR as to why.

Reply:
Do we know why `vaet` is faster and has less variance?

Reply:
No. Conjectures only.

Reply:
Got it. I wonder if we should add the fact that we don't know why it's better (and maybe the conjectures) to the comment in the code. I think our future selves (and other future team members) would thank us for that